packages/zoe/src/zoe.js

```diff
@@ -49,9 +50,11 @@ const makeZoe = (additionalEndowments = {}) => {
     // Resolve the payout promises with the payouts
     const pursePs = issuerTable.getPursesForIssuers(issuers);
     for (const offer of offers) {
-      const payout = offer.amounts.map((amount, j) =>
-        E(pursePs[j]).withdraw(amount, 'payout'),
-      );
+      const payout = {};
+      offer.amounts.forEach((amount, j) => {
+        payout[roleNames[j]] = E(pursePs[j]).withdraw(amount);
```

Comment:
The fact that you're updating this with assignment is a great example of why we needed to restrict the roleNames. If a roleName could possibly be `'__proto__'` this would do the wrong thing. We need to  ensure that all role names are validated before use.